x/concentrated-liquidity/lp.go

```diff
@@ -394,63 +402,64 @@ func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId
 // Updates ticks and pool liquidity. Returns how much of each token is either added or removed.
 // Negative returned amounts imply that tokens are removed from the pool.
 // Positive returned amounts imply that tokens are added to the pool.
+// If the lower and/or upper ticks are being updated to have zero liquidity, a boolean is returned to flag the tick as empty to be deleted at the end of the withdrawPosition method.
 // WARNING: this method may mutate the pool, make sure to refetch the pool after calling this method.
-func (k Keeper) UpdatePosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick, upperTick int64, liquidityDelta sdk.Dec, joinTime time.Time, positionId uint64) (sdk.Int, sdk.Int, error) {
+func (k Keeper) UpdatePosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick, upperTick int64, liquidityDelta sdk.Dec, joinTime time.Time, positionId uint64) (sdk.Int, sdk.Int, bool, bool, error) {
```

Comment:
Why don't we just have this method deal with deleting the tick? Why pass it up?

Reply:
Ah yeah we couldn't do this in initOrUpdateTick because the tick info is still needed for initOrUpdatePosition but I think we can do it in UpdatePosition instead of passing it one level higher, checking to be sure and then will make the change.

Reply:
@ValarDragon It needs to be passed up for the following reason:

We collect spread rewards when withdrawing an entire position https://github.com/osmosis-labs/osmosis/blob/465399ca3bda7c132eea78c99f82a8ef88db0028/x/concentrated-liquidity/lp.go#L236-L239

This calls prepareClaimableSpreadRewards https://github.com/osmosis-labs/osmosis/blob/465399ca3bda7c132eea78c99f82a8ef88db0028/x/concentrated-liquidity/spread_rewards.go#L175-L178

This calculates the spreadRewardGrowthOutside via the ticks https://github.com/osmosis-labs/osmosis/blob/465399ca3bda7c132eea78c99f82a8ef88db0028/x/concentrated-liquidity/spread_rewards.go#L247-L252

If we delete tick info earlier, this will output an incorrect amount

Reply:
I see